tasks/new_e2e_tests.py

```diff
@@ -95,3 +101,83 @@ def run(ctx, profile="", tags=[], targets=[], configparams=[], verbose=True, cac
     if some_test_failed:
         # Exit if any of the modules failed
         raise Exit(code=1)
+
+
+@task
+def clean(_):
+    """
+    Clean any environment created with invoke tasks or e2e tests
+    """
+    if not _is_local_state(_get_pulumi_about()):
+        print("Cleanup supported for local state only, run `pulumi login --local` to switch to local state")
+        return
+
+    print("🧹 Clean up lock files")
+    lock_dir = os.path.join(Path.home(), ".pulumi", "locks")
+
+    for entry in os.listdir(Path(lock_dir)):
+        subdir = os.path.join(lock_dir, entry)
+        for filename in os.listdir(Path(subdir)):
+            file_path = os.path.join(subdir, filename)
+            if os.path.isfile(file_path) and filename.endswith(".json"):
+                os.remove(file_path)
+                print(f"🗑️ Deleted lock: {file_path}")
+
+    print("🧹 Clean up stacks")
+    stacks = _get_existing_stacks()
```

Comment:
I am not sure why we should clean-up all stacks to `unlock` a task. Given the amount of time it happens, I'm not sure it's nice to delete all tasks.

Reply:
Should have mentioned in the PR description: this script is meant to clean up any leftover, but I can see how it could be 2 scripts, actually. When stopping a test in the middle, we might want to remove only locks. But when running a test after 24h from the first run, with the cleanup job doing its, well, job on the CI, we might also want to cleanup stacks. This is something devs were struggling with during the workshop.